### ISSUE TITLE
Slow entry creation and edit when we have a lot of assets

### ISSUE THREAD

**Issue by vasilisdotme** [statamic/cms#1722]
## Bug Description
When we create or edit an entry (even if the blueprint have only a title) takes between 12 to 20 seconds to complete loading. Data saved to mark down immediately. 

## How to Reproduce
On default assets container add 17.000 or more small images and then try to create or edit an entry

## Extra Detail

[Here](https://github.com/vasilisdotme/reproduce-statamic-assets-bug) you can find a repo with enough assets to reproduce the problem

## Environment

**Statamic version:** 3.0.0-beta.26

**PHP version:** 7.3.17

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by jasonvarga** [comment]
Are you using Amazon S3?

**Comment by vasilisdotme** [comment]
No, all assets are locally saved

**Comment by jasonvarga** [comment]
Thanks, and just clicked through to your example repo:

![](https://y.yarn.co/42415e43-41ec-41d2-aa51-878c0359db94_text_hi.gif)

**Comment by ryangittings** [comment]
Also seeing this! Lots of assets for a photography based site, seeing pages load very slowly.

**Comment by 1stevengrant** [comment]
I see slowness in saving an entry - even after I introduced the fix in #1477 

I don't have a lot of content or a lot of assets either - just takes too long to save and this is local via Valet.

https://d.pr/i/xzYPx1

**Comment by jasonvarga** [comment]
Figured out the cause of this is because Statamic is updating the search indexes when you save the entry. If assets are included ([and they are by default](https://github.com/statamic/statamic/blob/master/config/statamic/search.php#L30)) then it ends up looping over all the assets.

The checks that it needs to do can be smarter.

A workaround for this is to exclude assets from the index (update `searchables` below):

``` php
    'indexes' => [

        'default' => [
            'driver' => 'local',
            'searchables' => ['collection:*', 'taxonomy:*', 'users'],
            'fields' => ['title'],
        ],
]
```
